HpToolsLauncher/TestRunners/ApiTestRunner.cs

```diff
@@ -72,7 +72,36 @@ public TestRunResults RunTest(TestInfo testinf, ref string errorReason, RunCance
             ConsoleWriter.ActiveTestRun = runDesc;
             ConsoleWriter.WriteLine(DateTime.Now.ToString(Launcher.DateFormat) + " Running: " + testinf.TestPath);
 
-            runDesc.ReportLocation = testinf.TestPath;
+            // check if the report path has been defined
```

Comment:
Extract the if functionality to avoid code duplicity in ApiTestRunner and GuiTestRunner